common/src/main/templates/io/netty/util/collection/KObjectHashMap.template

```diff
@@ -608,7 +606,7 @@ public class @K@ObjectHashMap<V> implements @K@ObjectMap<V> {
             if (nextIndex == -1) {
                 scanNext();
             }
-            return nextIndex < keys.length;
+            return nextIndex != values.length;
```

Comment:
I believe so (correct me if I'm missing something). we never reduce the size of `values` and increment this index in `scanNext()` until we hit `values.length`. We may set `nextIndex` to the `prevIndex` but that is a previously accessed index which must be `< values.length`